File: 2025/Company/Company.Web/Modules/ServerTypes/Company/EmailAddressForm.ts
import { LookupEditor, StringEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface EmailAddressForm {
    BusinessEntityId: LookupEditor;
    EmailAddress: StringEditor;
    IsActive: IntegerEditor;
    TenantId: IntegerEditor;
}

export class EmailAddressForm extends PrefixedContext {
    static readonly formKey = 'Company.EmailAddress';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmailAddressForm.init)  {
            EmailAddressForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = IntegerEditor;

            initFormType(EmailAddressForm, [
                'BusinessEntityId', w0,
                'EmailAddress', w1,
                'IsActive', w2,
                'TenantId', w2
            ]);
        }
    }
}

File: 2025/Company/Company.Web/Modules/Company/EmailAddress/EmailAddressForm.cs
using Serenity.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.EmailAddress")]
[BasedOnRow(typeof(EmailAddressRow), CheckNames = true)]
public class EmailAddressForm
{
    public int BusinessEntityId { get; set; }
    public string EmailAddress { get; set; }
    public short IsActive { get; set; }
    public int TenantId { get; set; }
}